src/routes/courses.ts

```diff
@@ -40,15 +43,51 @@ router.get('/courses', (req, res) => {
     .catch((error: any) => res.status(500).json(error));
 });
 
-router.post('/courses', (req, res) => {
+router.get('/courses/:term?/:subject?/:code?', (req, res) => {
+  const term = req.params.term;
+  const subject = req.params.subject;
+  const code = req.params.code;
+  let query;
```

Comment:
This can be simplified a bit. Add the following interface at the top of the file:
```
interface Query {
  subject?: string;
  code?: string;
  term?: string;
}
```

The if-else statement can be simplified to:
```
const query: Query = {};

  if (term) query.term = term;
  if (subject) query.subject = subject;
  if (code) query.code = code;

```